### PR TITLE
Stricter content length in ASNObject

### PR DESCRIPTION
**Pull request by paragonie-security**
Also fixes a PHP 8 deprecation

### CODE REVIEW

File: lib/ASN1/Universal/OctetString.php [hunk 1]
```diff
@@ -47,18 +47,27 @@ public function getType()
...
+        if (strlen($value) & 1) {
+            $value = '0' . $value;
+        }
         $result = '';
 
         //Actual content
         while (strlen($value) >= 2) {
             // get the hex value byte by byte from the string and and add it to binary result
-            $result .= chr(hexdec(substr($value, 0, 2)));
+            $result .= @chr(hexdec(substr($value, 0, 2)));
```
Comment (paragonie-security) [inline lib/ASN1/Universal/OctetString.php:70]: We may want to replace `chr($val)` with `pack('C', $val)`.

https://paragonie.com/blog/2017/02/cryptographically-secure-php-development#chr

**Comment by frederikbosch** [comment]
For now I did not replace `chr($val)` with `pack('C', $val)`. To be honest, I am no security expert. I forked this because I thought at least someone should maintain the library. If you believe the `pack('C', $int)` is better, please create a PR for this. My only concern with the PR was the use of the `@` sign. I always try to avoid it. When I am going to revise the library, I will probably remove it.

**Comment by paragonie-security** [comment]
There's some weird behavior that this library seems to be relying on. That code in question seems to expect hex data, but sometimes operates over raw bytes. If you "fix" this bug, you bet a bunch of test failures. I suspect there be dragons here.

**Comment by frederikbosch** [comment]
That's exactly why I also chose to leave this for now. First, I am going to make this library use strict types and add phpstan. Then let's see which other optimizations and fixes we can introduce.

**Pull request merged**